template-base-3-typescript/template/src/hooks/Auth0.ts

```diff
@@ -1,21 +1,63 @@
 import { useDispatch } from 'react-redux';
-import { useEffect } from 'react';
+import { useCallback, useEffect, useMemo } from 'react';
 import { useAuth0 } from '@auth0/auth0-react';
 import { setCredentials } from '@carto/react-redux';
+import { initialState } from 'store/initialStateSlice';
+import { useSearchParams } from 'react-router-dom';
+
+const ACCOUNT_FULFILLED_PARAM = 'accountFulfilled';
+const buildAccountFulfilledUri = () =>
+  `${window.location.origin}?${ACCOUNT_FULFILLED_PARAM}=true`;
 
 export default function useAuth() {
-  const { isAuthenticated, getAccessTokenSilently } = useAuth0();
+  const { isAuthenticated, getAccessTokenSilently, user, loginWithRedirect } = useAuth0();
+  const [searchParams] = useSearchParams();
   const dispatch = useDispatch();
 
+  const accountsUrl = initialState.accountsUrl;
+  const organizationId = initialState.oauth?.organizationId;
+  const namespace = initialState.oauth?.namespace;
+
+  const accountHasBeenFulfilled = searchParams.has(ACCOUNT_FULFILLED_PARAM);
+
+  const getAccessToken = useCallback(async () => {
+    let accessToken = await getAccessTokenSilently();
+    dispatch(setCredentials({ accessToken }));
+  }, [dispatch, getAccessTokenSilently]);
+
+  const userMetadata = useMemo(() => {
+    if (!user) return;
+    return user[`${namespace}user_metadata`];
+  }, [user, namespace]);
+
+  const redirectAccountUri = useMemo(() => {
+    return `${accountsUrl}${organizationId ? `sso/${organizationId}` : ''}`;
+  }, [accountsUrl, organizationId]);
+
   useEffect(() => {
-    if (isAuthenticated) {
-      const getAccessToken = async () => {
-        let accessToken = await getAccessTokenSilently();
-        dispatch(setCredentials({ accessToken }));
-      };
+    if (isAuthenticated && userMetadata) {
       getAccessToken();
+    } else if (isAuthenticated) {
+      if (accountHasBeenFulfilled) {
```

Comment:
Not following why you need to trust the URL to see if the user has completed the account. I think it's possible by checking the metadata only.

Reply:
Something like this is not working? 
```javascript
 if (isAuthenticated && userMetadata) {
      getAccessToken();
    }
else if (isAuthenticated && !userMetadata) {
    / const searchParams = new URLSearchParams({
          redirectUri: buildAccountFulfilledUri(),
        });
        // Redirect to: accounts-www for signup on cloud-native
        window.location.href = `${redirectAccountUri}?${searchParams}`;
}
else {
   loginWithRedirect();
}
```

Reply:
No, to get the metadata you need to request a new token (with loginWithRedirect). As you were loged in before having the user_metadata you need to check the url and log in again.

Reply:
🤔